### DISCUSSION TITLE
Persistent token cache

### DISCUSSION THREAD

**Discussion by stefanodriussi** [pnp/pnpcore#400] [General] [answered]
I'm working on and Excel Addin and I need to connect to SharePoint in order to upload files on specific folders. I can successfully setup and connect using PnP.Core and PnP.Auth . Authentication flow allows me to pick the account and then all requests go through. However, whenever the application (Excel in this case) is restarted, the in memory token cache is wiped clean and the user has to go through the auth popup again.
I've followed [this](https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/wiki/token-cache-serialization) example and implemented my own custom `InteractiveAuthenticationProvider` that it's merely the same from this library with the additional binding between `IPublicClientApplication` and `MsalCacheHelper`. It works, tokens are cached and reused after application restart. What I'm trying to understand is first if It's the correct way or if there are any security implications using this approach and second if there's already something in place for this kind and I'm simply missing it.

**Comment by jansenbe** [comment, marked as answer]
@stefanodriussi : your approach is good, we base ourselves on what MSAL provides but added the flexibility to plug in custom auth providers, like you did. Would you like to contribute back your implementation as a community sample? We could highlight your sample over here: https://pnp.github.io/pnpcore/demos/README.html.

**Reply by stefanodriussi** [reply]
Sure, will share it. Would a simple console app work as well ? Just asking because VSTO addins are not exactly what I would define as a good starting point in order to explain something else 😅

**Reply by jansenbe** [reply]
Console app would be great. Thanks for sharing it!

**Reply by stefanodriussi** [reply]
Done !
You can find my sample here 👎 
https://github.com/stefanodriussi/Demo.PersistentTokenCache

**Reply by jansenbe** [reply]
Sorry @stefanodriussi , for some reason I never noticed this. I'll create an issue to track the work to make this into a sample. Thanks a ton for sharing!
